src/main/java/org/folio/dataimport/handlers/actions/CreateInvoiceEventHandler.java

```diff
@@ -259,20 +259,26 @@ private Future<Map<Integer, PoLine>> getAssociatedPoLinesByPoLineNumber(Map<Inte
 
 
   private Future<Map<Integer, PoLine>> getAssociatedPoLinesByRefNumbers(Map<Integer, List<String>> refNumberList, RequestContext requestContext) {
-    List<Future<Pair<Integer, PoLine>>> futures = new ArrayList<>();
-    Semaphore semaphore = new Semaphore(MAX_PARALLEL_SEARCHES, Vertx.currentContext().owner());
-
+    if (refNumberList.isEmpty()) {
```

Comment:
Let's use MapUtils.isNotEmpty, I personally use it in all places to not thinking can list or map be null or not